main/res/xml/preferences_map.xml

```diff
@@ -79,6 +79,11 @@
             android:summary="@string/settings_info_hillshading"
             android:title="@string/settings_info_hillshading_title"
             app:iconSpaceReserved="false" />
+        <CheckBoxPreference
+            android:defaultValue="false"
+            android:key="@string/pref_maphillshading"
+            android:title="@string/settings_hillshading_enable"
+            app:iconSpaceReserved="false" />
         <Preference
```

Comment:
You could make the other hillshading options depending on the master switch, so that they get greyed out if hillshading is disabled, see `android:dependency="@string/pref_maphillshading"`

Reply:
implemented with #13658